plugins/destination/postgresql/client/list_tables.go

```diff
@@ -88,22 +88,18 @@ func (c *Client) listTables(ctx context.Context) (schema.Tables, error) {
 			})
 		}
 		table := tables[len(tables)-1]
-		// Note: constraints always have a name in PostgreSQL.
-		// However, we want not only to store the info about the constraint name,
-		// but also the fact that we saw such table.
-		switch pkName {
-		case "":
-			// We still store the fact that we saw the table
-			if _, ok := c.pgTablesToPKConstraints[tableName]; !ok {
-				// Just store the empty string.
-				// This will indicate 2 things:
-				// 1. We saw the table with this name
-				// 2. If this is still empty on insert, the table in the database doesn't have a PK constraint
-				c.pgTablesToPKConstraints[tableName] = ""
-			}
-		default:
-			c.pgTablesToPKConstraints[tableName], table.PkConstraintName = pkName, pkName
+
+		// We always want to record that we saw the table, even if it doesn't have a PK constraint.
+		if _, ok := c.pgTablesToPKConstraints[tableName]; !ok {
+			c.pgTablesToPKConstraints[tableName] = &pkConstraintDetails{}
+		}
+		if pkName != "" {
+			c.pgTablesToPKConstraints[tableName].name = pkName
+			// we want to store any current column that is part of the PK
+			c.pgTablesToPKConstraints[tableName].columns = append(c.pgTablesToPKConstraints[tableName].columns, columnName)
+			table.PkConstraintName = pkName
```

Comment:
```suggestion
		entry, ok := c.pgTablesToPKConstraints[tableName]
		if !ok {
			entry = new(pkConstraintDetails)
			c.pgTablesToPKConstraints[tableName] = entry
		}
		if pkName != "" {
			entry.name = pkName
			// we want to store any current column that is part of the PK
			entry.columns = append(entry.columns, columnName)
			table.PkConstraintName = pkName
```

Reply:
Done in 1ee5ff6ca48caa7ae4bf950f406f026c85f54f70